teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorKeysOptions.java

```diff
@@ -122,8 +133,11 @@ public void configure(TekuConfiguration.Builder builder) {
                     convertToPath(validatorExternalSignerTruststorePasswordFile)));
   }
 
-  private List<String> parseValidatorExternalKeys() {
-    return validatorExternalSignerPublicKeys.stream().filter(StringUtils::isNotBlank).toList();
+  private List<String> parseValidatorExternalKeys(final boolean allowNoLoadedKeysEnabled) {
+    if (allowNoLoadedKeysEnabled) {
+      return validatorExternalSignerPublicKeys.stream().filter(StringUtils::isNotBlank).toList();
+    }
+    return validatorExternalSignerPublicKeys;
```

Comment:
At this point, we could already check if the list is empty and throw an error when `--allow-no-loaded-keys-enabled=false` and `validatorExternalSignerPublicKeys` is empty.